### PR TITLE
add darwin arm64 build, bump to Go 1.17.6

### PR DESCRIPTION
**Pull request by iameli**
This PR adds Darwin cross-compilation support for supporting both legacy amd64 and M1 arm64 macs.

All of the pieces here — install_ffmpeg.sh, Makefile, and upload_build.sh, are controlled through the $GOARCH environment variable. Should make it easy. If we ever support full cross-compilation, controlling it through $GOOS makes sense to me too.

**This also includes a version bump to Go 1.17.6**. There was a specific [compilation bug in 1.15.5](https://github.com/golang/go/issues/42565) and Go didn't support darwin-arm64 targets until Go 1.16. But if we're upgrading anyway I figure it makes sense to go to the latest and greatest.

### CODE REVIEW

File: install_ffmpeg.sh [hunk 1]
```diff
@@ -5,11 +5,22 @@ set -ex
...
 NPROC=${NPROC:-$(nproc)}
+EXTRA_CFLAGS=""
+EXTRA_LDFLAGS=""
+EXTRA_FFMPEG_FLAGS=""
 
 if [[ $ARCH == "arm64" ]] && [[ $(uname) == "Darwin" ]]; then
   # Detect Apple Silicon
   IS_M1=1
 fi
+if [[ $ARCH == "x86_64" ]] && [[ $(uname) == "Darwin" ]] && [[ "${GOARCH:-}" == "arm64" ]]; then
```
Comment (hjpotter92) [inline install_ffmpeg.sh:16]: is the `$ARCH` check needed here?

Comment (iameli) [reply install_ffmpeg.sh:16]: I think the only consequence would be that this block would run during native M1 ARM64 builds, resulting in ffmpeg having `--enable-cross-compile` enabled even though it doesn't need it. I'm not 100% sure what that flag does.

**Review by hjpotter92 — APPROVED** [review]
lgtm!

**Comment by hjpotter92** [comment]
> **This also includes a version bump to Go 1.17.6**. There was a specific [compilation bug in 1.15.5](https://github.com/golang/go/issues/42565) and Go didn't support darwin-arm64 targets until Go 1.16. But if we're upgrading anyway I figure it makes sense to go to the latest and greatest.

Should this also bump version in `go.mod`?

**Comment by iameli** [comment]
> Should this also bump version in go.mod?

My understanding is that field should reflect the most recent version of Go we'd expect go-livepeer to work with — our CI after this PR is using 1.17 but we'd expect things to work since the last time there was a breaking change. Helps with people pulling us in as a library and whatnot.

**Comment by iameli** [comment]
Held off on this until Confluence landed, merging now so we can get a darwin-arm64 build in the next semver release 😃

**Pull request merged**